src/components/menus/content-contextual-menu.js

```diff
@@ -380,6 +383,43 @@ const ContentContextualMenu = (props) => {
         );
     }, [isUserAllowed, selectedElements]);
 
+    const allowsDownloadCase = useCallback(() => {
+        //if selectedElements contains atleast one case
+        return selectedElements.some(
+            (element) => element.type === ElementType.CASE
+        );
+    }, [selectedElements]);
+
+    /**
+     * Downloads the selected cases as a file.
+     * @function
+     * @name handleDownloadCase
+     * @returns {void}
+     */
+    const handleDownloadCase = useCallback(() => {
+        //for each selectedElements , filter cases and return their uuid and subtype
+        const elementsToDownload = selectedElements
+            .filter((element) => element.type === ElementType.CASE)
+            .map((element) => ({
+                elementUuid: element.elementUuid,
+                subtype: element.subtype,
+            }));
+
+        elementsToDownload.forEach(async (element) => {
+            const result = await downloadCase(element.elementUuid);
+            const name = await getOriginalName(element.elementUuid);
+            const blob = await result.blob();
+            const href = window.URL.createObjectURL(blob);
+            const a = linkRef.current;
+            a.download = name;
+            a.href = href;
+            a.click();
+            a.href = '';
+        });
+    }, [selectedElements]);
```

Comment:
It would be nice to use the same way as we already do in "network-modification-tree-pane.js" in gridstudy-app to download the case

Reply:
Hello ,
in my case , i don't have a new pop-up, so i don't have anything to close. 
when the user click on download , it automatically download all selected cases

Reply:
Not talking about pop-up i'm talking about the menu, when you click on download the menu shoudn't stay open

Reply:
sorry for the delay.
yes you are right , the menue does not close when we click on download.  i fixed it.

Reply:
@AbdelHedhili for some reason in my case it doesn't work in my case because i recieve a blob from the API. i managed to download it but i didn't find a way to give it a name.
in study server the API return a url. and not a blob 

from what i understod from the documentation, iframe are used to ["to display a web page within a web page"](https://www.w3schools.com/html/html_iframe.asp) .

if this remark is not bloking for you i suggest to keep <a> tag

Reply:
ok let's not overcomplicate things since it works that way let's keep it.